### DISCUSSION TITLE
Question about Modules and providers

### DISCUSSION THREAD

**Discussion by agomezlucena** [google/guice#1775] [Q&A]
Hi I have a question, is possible to inject dependecies in a provider method? something like this:
```java
public class MyModule extends AbstractModule {
    @Provider
    public String name(){
        return "alex";
    }
    @Provider
    @Inject
    public Optional<String> test ( String name ) {
        return Optional.ofNullable( name );
    } 
}
```

**Comment by lukesandberg** [comment]
I believe you mean an `@Provides` method? if so the answer is yes.  You can inject arbitrary dependencies into an `@Provides` method just like you can into an `@Inject` constructor
